AEPMessaging/Sources/Messaging.swift

```diff
@@ -154,44 +169,198 @@ public class Messaging: NSObject, Extension {
 
-        let data: [String: Any] = [
+        // add a `data` object to the request specifying the format desired in the response from XAS
+        eventData[MessagingConstants.DATA.Key.DATA] = [
             MessagingConstants.DATA.AdobeKeys.NAMESPACE: [
                 MessagingConstants.DATA.AdobeKeys.AJO: [
-                    MessagingConstants.DATA.AdobeKeys.INAPP_RESPONSE_FORMAT: 2 // Int representing latest in-app format. Absence implies old in-app format.
+                    MessagingConstants.DATA.AdobeKeys.INAPP_RESPONSE_FORMAT: MessagingConstants.XDM.Inbound.Value.IAM_RESPONSE_FORMAT
                 ]
             ]
         ]
-        eventData[MessagingConstants.DATA.Key.DATA] = data
-
-        let event = Event(name: MessagingConstants.Event.Name.RETRIEVE_MESSAGE_DEFINITIONS,
-                          type: EventType.edge,
-                          source: EventSource.requestContent,
-                          data: eventData)
+        
+        // add a `request` object so we get a response event from edge when the propositions stream is closed for this event
+        eventData[MessagingConstants.XDM.Key.REQUEST] = [
+            MessagingConstants.XDM.Key.SEND_COMPLETION: true
+        ]
+        // end construction of event data
+
+        let newEvent = event.createChainedEvent(name: MessagingConstants.Event.Name.RETRIEVE_MESSAGE_DEFINITIONS,
+                                                type: EventType.edge,
+                                                source: EventSource.requestContent,
+                                                data: eventData)
+
+        // create entries in our local containers for managing streamed responses from edge
+        beginRequestFor(newEvent, with: requestedSurfaces)
+
+        // dispatch the event and implement handler for the completion event
+        MobileCore.dispatch(event: newEvent, timeout: 10.0) { responseEvent in
+            // responseEvent is the event dispatched by Edge extension when a request's stream has been closed
+            guard let responseEvent = responseEvent,
+                let endingEventId = responseEvent.requestEventId else {
+                // response event failed or timed out, need to remove this event from the queue
+                self.requestedSurfacesForEventId.removeValue(forKey: newEvent.id.uuidString)
+                
+                Log.warning(label: MessagingConstants.LOG_TAG, "Unable to run completion logic for a personalization request event - unable to obtain parent event ID")
+                return
+            }
+            
+            // dispatch an event signaling messaging extension needs to finalize this event
+            // it must be dispatched to the event queue to avoid a race with the events containing propositions
+            let processCompletedEvent = responseEvent.createChainedEvent(name: MessagingConstants.Event.Name.FINALIZE_PROPOSITIONS_RESPONSE,
```

Comment:
I believe we can avoid dispatching this internal event by creating an event queue (`OperationOrderer<Event>("<messaging-queue>")`) in the extension. We can use it to order and process the Edge contentComplete event only after all personalization decision events for the same requestEventId are processed. I'll try it out.